backend/gsr_booking/api_wrapper.py

```diff
@@ -2,24 +2,31 @@
 from abc import ABC, abstractmethod
 from enum import Enum
 from random import randint
+from typing import TYPE_CHECKING, Any
 
 import requests
 from bs4 import BeautifulSoup
 from django.conf import settings
 from django.contrib.auth import get_user_model
-from django.db.models import F, Prefetch, Q, Sum
+from django.db.models import F, Prefetch, Q, QuerySet, Sum
 from django.db.models.functions import Coalesce
 from django.shortcuts import get_object_or_404
 from django.utils import timezone
 from requests.exceptions import ConnectionError, ConnectTimeout, ReadTimeout
 
-from gsr_booking.models import GSR, GroupMembership, GSRBooking, Reservation
+from gsr_booking.models import GSR, Group, GroupMembership, GSRBooking, Reservation
 from gsr_booking.serializers import GSRBookingSerializer, GSRSerializer
 from utils.errors import APIError
 
 
-User = get_user_model()
+if TYPE_CHECKING:
```

Comment:
I'm going to guess its because Django makes us do `get_user_model()`? 

I feel like its clunky to have to put this at the top of all the files.. Maybe we can put this in a file and just import it everywhere? I'm starting to wonder how other Django projects do this cleanly

Reply:
it's cause get_user_model() is a function, so it gets annoyed when we try to declare its type. but yes put it in another file under utils/types